### DISCUSSION TITLE
Add ability to reduce or disable stereo

### DISCUSSION THREAD

**Discussion by SonoSooS** [ppy/osu#21380] [Ideas]
Haven't updated lazer since half a year, but now the results screen has positional audio which is so strong that it makes me sick.

I know that some stereo effects have been existing in the game since basically forever, but they are usually really quiet, and not that strong, so I've tried my best to trigger them the least amount.
But since it's really annoying to wait out the long win music of the skin, and it's a difficult job to erase the win sound from all of the skins I have (especially with lazer's folder structure), it is not something I'm really looking forward to do.

Would it be possible to add an accessibility option to reduce the strength of these effects, or be able to disable them all together?
There is "hitsound stereo separation", but it doesn't seem to affect anything else.

And while on the topic of stereo, it would be great if it was possible to disable stereo in the beatmap's music as well. Some of them have extremely nauseating stereo effects.

**Comment by Doksha** [comment]
I feel like this should simply be added to the functionality of the hitsound stereo separation slider. (ie, that slider would control stereo throughout the whole game)

**Comment by peppy** [comment]
You can disable stereo at a windows level, would that be a solution you could accept?

**Reply by Doksha** [reply]
the only problem is that that applies to all apps, meaning if you forget to disable it you are left wondering why everything sounds like garbage. It works just fine, but having to enable mono audio every time you play osu would be a little annoying.

**Reply by peppy** [reply]
I'd argue that if you have an issue with stereo sound (to the point it makes you feel bad) it would probably affect you across all apps and this is what you want.

**Reply by SonoSooS** [reply]
I don't have that option, so I made my own program to do it. The problem is that it adds so much latency that it simply just doesn't work with the game.

Also not all stereo content I "consume" is bad. Most of  the times the thing that makes me sick is unbalanced audio (that is, panning). There are some other phase trickery effects which don't directly involve panning which make me sick, but those are quite rare.

I just need these hard panning effects reduced (or preferably, completely turned off). I get that this is a stylistic choice, but in previous updates this was not present, and it was added with no way to disable it, so it's extremely painful now that there are even more loud hard pans with no way to disable them (without of course finding them in code, and compiling a dev build).

**Reply by Doksha** [reply]
what do you mean by that? what operating system are you using?
windows has mono audio as an accessibility option

**Reply by SonoSooS** [reply]
Windows 10, but a super outdated version which lacks the "Mono audio" option and some other settings.

Edit: but even if I had, I still need to use my program for audio filtering because it has configurable high and low pass filters to filter out certain frequencies to make it easier to understand speech, but also filter out mouth noises.

**Reply by Doksha** [reply]
have you tried equalizer APO? you can use the copy channels feature to emulate mono audio with an easy toggle, and you can add however many pass filters you want

**Reply by SonoSooS** [reply]
I tried it, but even if I manage to install it at all, I won't have audio output, so I can't use it.
I also made my own APO in response, but it experiences the same problem, no audio out. The callback doesn't even get called, so there has to be something weird going on.

I blame Realtek, they probably did something weird again. Not only that, their driver randomly crashes, and I have to restart audiosrv, or sometimes have to completely reboot to restore audio. In fact, sometimes audio doesn't even work on boot. It could very well be a driver issue, as I had to hunt for drivers a lot, as every driver I tried either has problems with the microphone, or the headphone output has various problems, like extreme bit-crush "quality", very quiet, or extremely noisy with loud pops. Not fun.
I won't even talk about the issues I have with this sound chip in Linux and macOS. I don't dare to touch any of my config again, it just works, after days of fiddling, driver hunting, and registry-crusade.

**Reply by Doksha** [reply]
you need to open the configurator (not configuration editor), click on the audio device (not the check mark) and select the troubleshooting options at the bottom, then select the dropdown on the right that says "install as LFX/GFX" and switch it to "install as SFX/MFX" (or sfx/efx)

**Reply by SonoSooS** [reply]
I am too afraid to try it anymore. This time I have too much to lose if I have to reinstall Windows again due to permanently broken audio subsystem.

~~I think I'll just try finding a cold-storage HDD with some free space where I could move some data so I could re-install the .NET Core SDK, and just edit out the panning support for the time being. I'm not getting unbanned anytime soon, so it doesn't matter if I'm using a dev build or a retail build.~~

~~But in the long run it would be great if I didn't have to deal with merge conflicts and wasted disk space just to play the game.~~

Edit: I forgot I can patch the game. Astonishing difference to have game-wide positional effects disabled! Feel much, MUCH better.
Still sub-optimal to have to patch the game, but still better than having to deal with merge conflicts or with the buggy .NET Core build system.

**Reply by smoogipoo** [reply]
I think it's unreasonable to expect users to disable system=wide settings in response to features in the game, especially one such as stereo audio... At the same time I don't think it needs an on/off setting, and instead we should be able to/need to tune the panning to a reasonable level.

**Reply by SonoSooS** [reply]
Well most stereo pan effects are already at a "reasonable" level, except when the cursor glitches out of bounds due to resolution change and I accidentally click, then it's almost a hard pan.
I put "reasonable" in quotes, because I'm so sensitive to this that I would just disable it outright, but if I wasn't sensitive to it, then it's actually strong enough to be noticable, but weak enough to be a small little effect to really appreciate, just as a nice touch.

Personally I would completely disable the pan effect on the results screen in particular, because I simply just don't like it as a headphone user. On speaker it would be definitely a really nice effect, but I think it would wear off quickly if you're the type to view your accuracy distribution often, like I do.

Considering how extremely rare this "illness" (or whatever this is, not even three neurologists I have visited knows what this could be) is, and how I'm seemingly the first one to ever report this, and how much resistance I'm getting on trying to work out a solution for this, this time I'll accept defeat, as peppy knows me already for being notorious at patching the game a lot anyway, so I can just keep patching it out once a new version comes.

Although even if my forfeiture is accepted, I would still love if something would be done about the results screen, like either completely disabling panning when expanding the accuracy chart, or having it be affected by the hitsound pan slider (mine is at 0%, obviously), or have a config value in game.ini which would directly pass that config value to the MinValue and MaxValue of the BindableDouble for the Balance effect (that's how the panning property (or field?) is called in code, sadly I have forgot the class for it though), so I could edit game.ini to completely disable all panning effects in the game, and then the settings UI would not need to have be littered with an option nobody but basically only me(?) would use.

**Reply by ghost** [reply]
Very likely you're not going to be the only one using this. I think many people would like a setting to adjust stereo effect strength.

I also think the effects on the beatmap results screen is a bit strong, I just never reported it
